Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/BlendGraphWidget.h

```diff
@@ -37,20 +37,29 @@ namespace EMStudio
     class BlendGraphMimeEvent : public GraphCanvas::GraphCanvasMimeEvent
     {
     public:
-        explicit BlendGraphMimeEvent(QString typeString, QString namePrefix);
+        AZ_RTTI(BlendGraphMimeEvent, "{AA7C8960-C7BA-4F26-B52B-97CF4AC3CB39}", GraphCanvas::GraphCanvasMimeEvent);
+        AZ_CLASS_ALLOCATOR(BlendGraphMimeEvent, AZ::SystemAllocator, 0);
+        static void Reflect(AZ::ReflectContext* context);
+
+        BlendGraphMimeEvent() = default;
+        explicit BlendGraphMimeEvent(const AZStd::string_view typeString, const AZStd::string_view namePrefix);
+
         bool ExecuteEvent(const AZ::Vector2& sceneMousePosition, AZ::Vector2& sceneDropPosition, const AZ::EntityId& sceneId) override;
 
-        QString GetTypeString() const;
-        QString GetNamePrefix() const;
+        AZStd::string GetTypeString() const;
+        AZStd::string GetNamePrefix() const;
+
+        static constexpr const char* BlendGraphMimeEventType = "animgraph/node-palette-mime-event";
+
     private:
-        QString m_typeString;
-        QString m_namePrefix;
+        AZStd::string m_typeString;
+        AZStd::string m_namePrefix;
     };
 
-    class BlendGraphNodePaletteTreeItem : public GraphCanvas::IconDecoratedNodePaletteTreeItem
+    class BlendGraphNodePaletteTreeItem : public GraphCanvas::DraggableNodePaletteTreeItem
     {
     public:
-        BlendGraphNodePaletteTreeItem(AZStd::string_view name, const QString& typeString, GraphCanvas::EditorId editorId, const AZ::Color& color);
+        BlendGraphNodePaletteTreeItem(const AZStd::string_view name, const QString& typeString, GraphCanvas::EditorId editorId, const AZ::Color& color);
```

Comment:
suggestion: We don't usually add `const` to the header/interface when the type is pass by value as it doesn't add anything and is redundant. It's fine in the .cpp/implementation to indicate to the reader (and compiler) the value won't change but here it can be removed

Reply:
Done